crates/matrix-sdk/src/sliding_sync.rs

```diff
@@ -794,7 +809,12 @@ impl SlidingSync {
                     pos,
                     room_subscriptions,
                     unsubscribe_rooms,
-                    extensions: extensions.lock_mut().take().unwrap_or_default(), // extensions are sticky, we pop them here once
+                    // extensions are sticky, so we could. pop them here once; they will get rebuilt again if needed
+                    // ...except this doesn't work: if this request fails for some reason (e.g. timing out),
+                    // then we don't know to re-add the sticky extension on the next request. So either we need to retry
+                    // and queue all requests as a matter of course - or we need to only pop extensions when we know this
+                    // req has succeeded. for now, we make extensions non-sticky to make things work reliably...
```

Comment:
If that's the case, can't we move the `take()` call after the requests is done? Perhaps that was the intention of the OP here.

Reply:
that's what i've done in the end.